components/coding/studentPortal/radix/TooltipComponent.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import {
+  Provider,
+  Root,
+  Trigger,
+  Portal,
+  Content,
+  Arrow,
+} from "@radix-ui/react-tooltip";
+
+interface TooltipComponentProps {
+  message: string;
+  active: boolean;
+}
+
+const TooltipComponent: React.FC<TooltipComponentProps> = ({
+  children,
+  message,
+  active,
+}) => {
+  return active ? (
+    <Provider>
+      <Root>
+        <Trigger asChild>{children}</Trigger>
+        <Portal>
+          <Content
+            className="data-[state=delayed-open]:data-[side=top]:animate-slideDownAndFade data-[state=delayed-open]:data-[side=right]:animate-slideLeftAndFade data-[state=delayed-open]:data-[side=left]:animate-slideRightAndFade data-[state=delayed-open]:data-[side=bottom]:animate-slideUpAndFade text-red-500 select-none rounded-[4px] bg-white px-[15px] py-[10px] text-[15px] leading-none shadow-[hsl(206_22%_7%_/_35%)_0px_10px_38px_-10px,_hsl(206_22%_7%_/_20%)_0px_10px_20px_-15px] will-change-[transform,opacity]"
```

Comment:
This is super confusing. What's happening here?

Reply:
I greatly modified the tailwind classes here.
I chose to have the tooltip always appear from the top and the therefore could eliminate the other three "data-...." for left/right/bottom.  

I also removed the shadow class and simplified it to "shadow-md ring-1 ring-gray-300".  It's not quite as nice, but is similiar.

Finally, to explain..."select-none" doesn't allow cursor clickable events.  leading-none removes any additional vertical space in the tooltip and sets the line height to the text size.  "will-change-[transform,opacity]" signals to the browser that there will be an animation change in opacity/transform will happen here in the tailwind.config.ts file " contentShow: {
          from: { opacity: 0, transform: 'translate(-50%, -48%) scale(0.96)' },
          to: { opacity: 1, transform: 'translate(-50%, -50%) scale(1)' },
        },"